src/rendering/rendering.zig

```diff
@@ -672,10 +682,12 @@ fn internalCollect(
 
     const context_source = comptime ContextSource.fromData(@TypeOf(data));
     const PartialsMap = map.PartialsMapType(@TypeOf(partials), options);
+    const Data = @TypeOf(data);
     const RenderEngine = RenderEngineType(
         context_source,
         @TypeOf(writer),
         PartialsMap,
+        Data,
```

Comment:
With the changes I made, a `DataRender` object has now 2 contexts:
- one for the user data type (native, JSON or ffi) 
- one for the global lambdas type (a native one (by the way, I have to specify that, currently it could be also JSON or ffi)).
A global lambda is only valid if the lambda signature function has:
- 1 `LambdaContext` parameter or
- 2 parameters with: 1 concrete type of the user data context + 1 `LambdaContext` parameter
So now the global lambda context needs to know the concrete type of the user data context.

So yes maybe we could remove the `context_source` parameter

Reply:
If we want to avoid passing the concrete type of the context here we need to think how I made things differently. If I choosed to pass the concrete type of the context here, this is for the `isValidLambdaFunction` problem I described some days ago.

As a potential solution that needs more thinking, maybe we could avoid passing the concrete type of the context if we solve this problem before https://github.com/batiati/mustache-zig/pull/24#discussion_r1700535816 
The concrete type if already contained in the `DataRender` object so maybe we can use it instead of this new parameter I added into `RenderEngineType()`. But it is only speculation. I am going to work on this possibility during the next days.

Reply:
Fixed by https://github.com/batiati/mustache-zig/pull/24/commits/f73ec988f7126b02e3e7478adb11884fa8f17026